Content/Bosses/VitricBoss/NPCs.ArenaBottom.cs

```diff
@@ -349,6 +338,19 @@ public override void DrawBehind(int index, List<int> behindNPCsAndTiles, List<in
 
 		public override void PostDraw(Color lightColor)
 		{
+			if (!loaded)
```

Comment:
Spawning this in a draw hook isnt great, even if it is being theoretically 'locked' by this loaded var, is there a reason for this being moved out of ai?

Reply:
it's in the draw hook since it's pure gores, dusts, and audios (which I think shouldn't be in AI where possible)

Reply:
if not this, then you need to put  Main.netMode != NetmodeID.Server all over wherever you're using dusts or gores. the server can handle vanilla ones but will kill NPCs or projectiles that use modded gores and dusts. though it would be better for the server to not even need to process vanilla gores and dusts

Reply:
My main concern with spawning dust in drawing hooks is that it is not in sync with the update clock, and higher framerates can cause more dust than expected to spawn.